firefox-ios/Storage/Rust/RustPlaces.swift

```diff
@@ -32,6 +32,7 @@ import struct MozillaAppServices.VisitTransitionSet
 public protocol BookmarksHandler {
     func getRecentBookmarks(limit: UInt, completion: @escaping ([BookmarkItemData]) -> Void)
     func getBookmarksTree(rootGUID: GUID, recursive: Bool) -> Deferred<Maybe<BookmarkNodeData?>>
+    func countBookmarksInTrees(folderGuids: [GUID]) -> Deferred<Maybe<Int>>
```

Comment:
i think we should remove Deferred from our code base since i guess it is legacy code. what do you think to refactor this either with an async method if possible otherwise a closure ? 👀